p2p/src/conn.rs

```diff
@@ -98,12 +103,7 @@ impl<'a> Message<'a> {
 		while written < len {
 			let read_len = cmp::min(8000, len - written);
 			let mut buf = vec![0u8; read_len];
-			read_exact(
-				&mut self.stream,
-				&mut buf[..],
-				time::Duration::from_secs(10),
-				true,
-			)?;
+			self.stream.read_exact(&mut buf[..])?;
```

Comment:
Is this now controlled by `IO_TIMEOUT` (i.e. 1 second) ?  if the buf can't be filled in 1 second, `read_exact` will return a TIMEOUT error. So, now the minimum speed is 8kByte per second, instead of old 800 Byte/s. 

Is my understanding correct?

Reply:
Not exactly, `read` (syscall) timeout occurs when we we can't read a single byte in 1 second. In our `read_exact` we didn't have a timeout for a single read, but for the entire operation (filling a buffer). In Rust `read_exact` we don't have any high-level timeouts but can control the timeout for a single read. 
So it's quite opposite, the previous impl was more demanding for connection speed, a new one will work for any connection faster than 1 byte/sec. A drawback that a peer's thread can be blocked by `read_exact` for a while on a slow connection, that what we can experience during shutdown.

Reply:
> read (syscall) timeout occurs when we we can't read a single byte in 1 second.

In the case of a peer's sending thread doesn't get the CPU resources for 1 second, does this `read_exact` will return a timeout error with part of buff filled?

Reply:
@hashmap   remind above question in case you missed it :-)

Reply:
Ah, thanks! Well, this is timeout for a single read operation which is basically syscall, I'm not an expert in preemptive scheduling but I think it's safe to assume that this situation is very unlikely.